distribution/sc-proxy-dex/src/proxy_pair.rs

```diff
@@ -561,13 +586,31 @@ pub trait ProxyPairModule: proxy_common::ProxyCommonModule {
     }
 
     #[view(getTemporaryFunds)]
+    fn get_temporary_funds(
+        &self,
+        address: &Address,
+    ) -> MultiResultVec<GenericEsdtAmountPair<Self::BigUint>> {
+        let vec: Vec<GenericEsdtAmountPair<Self::BigUint>> = self
+            .temporary_funds(address)
+            .iter()
+            .map(|x| {
+                let (key, amount) = x;
+                let (token_id, token_nonce) = key;
+                GenericEsdtAmountPair {
+                    token_id,
+                    token_nonce,
+                    amount,
+                }
+            })
+            .collect();
```

Comment:
You can `.collect()` directly into a `MultiResultVec`, without needing to use an intermediate `Vec`

Reply:
Done